File: src/main/java/xyz/chunkstories/api/entity/traits/TraitAnimated.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.entity.traits;

import xyz.chunkstories.api.animation.Animator;
import xyz.chunkstories.api.entity.Entity;

public abstract class TraitAnimated extends Trait {
	public TraitAnimated(Entity entity) {
		super(entity);
	}

	public abstract Animator getAnimatedSkeleton();
}

File: src/main/java/xyz/chunkstories/api/entity/traits/TraitAnimated.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.entity.traits

import xyz.chunkstories.api.animation.Animator
import xyz.chunkstories.api.entity.Entity

abstract class TraitAnimated(entity: Entity) : Trait(entity) {

    abstract val animatedSkeleton: Animator
}
